src/common/store/pages/audio-rewards/slice.ts

```diff
@@ -125,6 +137,38 @@ const slice = createSlice({
     ) => {
       const { status } = action.payload
       state.cognitoFlowStatus = status
+    },
+    fetchClaimAttestation: (
```

Comment:
oh hey just a thought, i wonder if this could be helpful: https://redux-toolkit.js.org/rtk-query/overview i think can help autogen the pending/success/error/retry states

Reply:
oh yeah definitely - this one got pretty verbose. I think we talked about this tool before, but might be worth bringing up again next prod sync!